packages/@lwc/engine-server/src/__tests__/fixtures/context-simple/modules/x/consumer/consumer.js

```diff
@@ -1,6 +1,6 @@
 import { LightningElement, wire } from 'lwc';
 import { WireAdapter } from '../../../wire-adapter';
 
-export default class Component extends LightningElement {
-  @wire(WireAdapter) foo;
+export default class ConsumerComponent extends LightningElement {
+  @wire(WireAdapter, { unused: '$definitelyUnused' }) foo;
```

Comment:
Wire adapters with config deserve more test coverage. I think I implemented it correctly but, because my time is constrained, I did not spend time thinking through the scenarios here and adding tests. Should probably be addressed in a follow-up WI.

Reply:
- https://github.com/salesforce/lwc/issues/4817